File: 0198-house-robber/0198-house-robber.py
class Solution(object):
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        rob, norob = 0, 0
        
        for num in nums:
            newRob = norob + num
            newNoRob = max(norob, rob)
            rob, norob = newRob, newNoRob
            
        return max(rob, norob)

File: 0198-house-robber/0198-house-robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        //Keep track of max amt when robbing/not robbing 
        int robHouse = 0; int notRob = 0;
        for (int i = 0; i < nums.size(); i++) {
            //update new maxes 
            int newRob = notRob + nums[i];
            int newNoRob = max(notRob, robHouse);
            
            //update the variables
            robHouse = newRob;
            notRob = newNoRob;
        }
        //return max of scenarios
        return max(robHouse, notRob);
    }
};